py/miniconf-mqtt/miniconf/discover.py

```diff
@@ -30,35 +33,30 @@ async def discover(
     Returns:
         A list of discovered client prefixes that match the provided filter.
     """
-    suffix = "/alive"
-
-    sub = {}
-
-    def on_subscribe(_client, mid, _qos, _props):
-        sub[mid].set_result(True)
-
     discovered = []
+    suffix = "/alive"
+    topic = f"{prefix}{suffix}"
 
-    def on_message(_client, topic, payload, _qos, _properties):
-        logging.debug("Got message from %s: %s", topic, payload)
-        if json.loads(payload):
-            discovered.append(topic[: -len(suffix)])
-
-    if isinstance(client, str):
-        client_ = Client(client_id="")
-        await client_.connect(client)
-        client = client_
-    client.on_subscribe = on_subscribe
-    client.on_message = on_message
-
-    fut = asyncio.get_running_loop().create_future()
     t_start = asyncio.get_running_loop().time()
-    sub[client.subscribe(f"{prefix}{suffix}")] = fut
-    await fut
-    t_rtt = asyncio.get_running_loop().time() - t_start
-    await asyncio.sleep(rel_timeout * t_rtt + abs_timeout)
-
-    client.unsubscribe(f"{prefix}{suffix}")
-    client.on_subscribe = lambda *_a, **_k: None
-    client.on_message = lambda *_a, **_k: None
+    await client.subscribe(topic)
```

Comment:
We still have `client` type variables as `str | Client`, but its now only `Client`

Reply:
Changed to just `Client` in 095c23edab750f2e4946c71d1e8065537472ccde